main/src/main/scala/org/clulab/processors/clu/CluProcessor.scala

```diff
@@ -235,47 +234,45 @@ class CluProcessor protected (
   /** Restores the correct case for all words in a given document */
   def restoreCase(doc: Document): Unit = {
     GivenConstEmbeddingsAttachment(doc, true).perform {
-      for(sent <- doc.sentences) {
-        // the case restoration model expects lower-case words as input                    
-        val loweredWords = sent.words.map(_.toLowerCase())
+      for (sent <- doc.sentences) {
+        // The case restoration model expects lower-case words as input.
+        val originalWords = sent.words
+        val loweredWords = originalWords.map(_.toLowerCase)
         val preLabels = mtlCase.predict(0, AnnotatedSentence(loweredWords), None, getEmbeddings(doc))
         val labels = casePostProcessing(loweredWords, preLabels)
-        assert(labels.size == loweredWords.size)
```

Comment:
There should be a stronger guarantee in `casePostProcessing` that this is the case.